vsintegration/src/FSharp.Editor/LanguageService/SingleFileWorkspaceMap.fs

```diff
@@ -74,7 +83,47 @@ type internal SingleFileWorkspaceMap(workspace: VisualStudioWorkspace,
 
         member _.OnAfterFirstDocumentLock(_, _, _, _) = VSConstants.E_NOTIMPL
 
-        member _.OnAfterSave(_) = VSConstants.E_NOTIMPL
+        member _.OnAfterSave(docCookie) = 
+            try
+                // This causes re-analysis to happen when a F# document is saved.
+                // We do this because FCS relies on the file system and existing open documents
+                // need to be re-analyzed so the changes are propogated.
+                // We only re-analyze F# documents that are dependent on the document that was just saved.
+                // We ignore F# script documents here.
+                let vsTextBuffer = rdt4.GetDocumentData(docCookie) :?> IVsTextBuffer
```

Comment:
Note OnAfterSave could be called for things that aren't text files. (Consider designers, even for non-F# projects.)

Reply:
I'm going to add more checks to make sure we do not do any work for things we don't care about.